File: 61.旋转链表.py
#
# @lc app=leetcode.cn id=61 lang=python3
#
# [61] 旋转链表
#
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def rotateRight(self, head: ListNode, k: int) -> ListNode:
        len = self.getListLength(head)
        
        print(k%len)
        # i = k%i

    def getListLength(self, head: ListNode) -> int:
        node = head
        len = 0
        while node:
            len = len+1
            node = node.next
        return len    
    

File: 61.旋转链表.js
/*
 * @lc app=leetcode.cn id=61 lang=javascript
 *
 * [61] 旋转链表
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var rotateRight = function(head, k) {
    let len = 0;
    const node = head;
    let pos = 0;
    while(node){
        len++;
        node = node.next;
    }
    console.log(len);
    pos = k%len;
    for (let index = 0; index < pos; index++) {
        const element = array[index];
        
    }
};
// @lc code=end

